lib/unleash/constraint.rb

```diff
@@ -92,6 +94,12 @@ def matches_constraint?(context)
         false
       end
 
+      # when the operator is NOT_IN and there is no data, return true. In all other cases the operator doesn't match.
+      return self.operator == :NOT_IN unless context.property?(self.context_name)
+
+      Unleash.logger.debug "Unleash::Constraint matches_context? value: #{self.value} context.get_by_name(#{self.context_name})" \
+      " #{context.get_by_name(self.context_name)} "
```

Comment:
should switch order here, and rename the method name within the log message / rewrite it to the log messages current context.

```suggestion
      Unleash.logger.debug "Unleash::Constraint matches_constraint?  value: #{self.value} constraint: #{self.operator} context.get_by_name(#{self.context_name})" \
      " #{context.get_by_name(self.context_name)} "

      # when the operator is NOT_IN and there is no data, return true. In all other cases the operator doesn't match.
      return self.operator == :NOT_IN unless context.property?(self.context_name)
```

Reply:
We cant log it before we validate that it is set. This is the reason the log is behind the check if property is set.

Reply:
why not?  `context.get_by_name()` should return nil if no value is found.


We would like to get this debug message for all operators, including the `:NOT_IN`, and even if the value is not in the context.

Also the node regarding having `matches_constraint?` instead of `matches_context?` within the log message is still valid.